docs/spec_draft.md

```diff
@@ -0,0 +1,124 @@
+ %2 = stablehlo.ceil  %arg : tensor<4x16xf32>
+ %3 = stablehlo.add %1, %2 : tensor<4x16xf32>
+ func.return %3 : tensor<4x16xf32>
+}
+```
+
+A program is executed by passing argument values to a given function and
+computing result values. Result values of a function are computed by evaluating
+the graph of ops rooted in the corresponding return op. The evaluation order is
+implementation-defined, as long as ops are evaluated before their uses. Possible
+execution orders of the above example program are `%1` → `%2` → `%3` → `return`
+or `%2` → `%1` → `%3` → `return`.
+
+## Constants
+
+The section describes the constants supported in StableHLO along with their
+syntax.
+
+  * **Integer Constants** Standard integers are constants of the integer type
+  (signed or unsigned). Negative numbers can be used with signed integer types.
+  * **Boolean Constants** "true" and "false" are both valid constants of the
+  `pred` type.
+  * **Floating point Constants** Floating point constants use standard decimal
+  notation, e.g., "123.421", exponential notation, e.g. "1.23421e+2", or a more
+  precise hexadecimal notation, e.g., "0x42f6d78d".
+  * **Complex Constants** Complex constants are represented as a pair of real
+  and imaginary values of `f32` or `f64` types .
+
+## Structure of an Op’s Specification
+
+The specification of an op comprises of the following components (in the order
+    described below)
+
+  * **Syntax**  Operation mnemonic and its signature.
+  * **Semantics** Semantics of the operation.
+  * **Operands** Meaning of operand(s) and their type(s).
+  * **Results** Meaning of the result(s) and the type(s).
+  * **Constraints** Type constraints on the operand(s), result(s).
+  * **Examples** Examples demonstrating the working of the op.
+
+## stablehlo.add
+
+`stablehlo.add(lhs, rhs) -> result`
+
+### Semantics
+
+Performs element-wise addition of two tensors `lhs` and `rhs` and produces a
+`result` tensor.  If an element-wise sum has an unsigned or signed overflow, the
+result is implementation defined and one of the followings:
+
+  * mathematical result modulo 2n, where n is the bit width of the result.
+  * saturation to 2^(n-1) - 1 (or -2^(n-1)) for signed overflow (or signed
+      underflow) and saturation to 2^n - 1 (or 0) for unsigned overflow (or
+        unsigned underflow).
+
+### Arguments
+
+| Operand Name(s) | Type |
+|-|-|
+| `lhs` | `tensor` of Integer, Floating point and Complex element types |
+| `rhs` | `tensor` of Integer, Floating point and Complex element types |
+
+### Results
+
+| Result Name | Type |
+|-|-|
+| `result` | `tensor` of Integer, Floating point, and Complex element types |
+
+### Constraints
+
+  * `lhs`, `rhs` have the same type.
+  * Supported shapes: all static shapes.
+  * `result` must have the same type as that of `lhs` (or `rhs`).
+
+### Examples
+
+```c
+// %x: [[1, 2], [3, 4]]
+// %y: [[5, 6], [7, 8]]
+%z = stablehlo.add %x, %y : tensor<2x2xf32>
+// %z: [[6, 8], [10, 12]]
+```
```

Comment:
In the near future, let's think about the best way of providing these examples. For example, can we just link to the interpreter implementation of the op, so that people can check out the test suite? Or, more generally, let's rename this section to "Implementation" and link to the source code? We shouldn't block this CL on that though, but let's open a ticket.

Reply:
done.